lib/device.ex

```diff
@@ -14,6 +13,7 @@ defmodule Onvif.Device do
                 :hardware_id,
                 :ntp,
                 :media_service_path,
+                auth_type: :no_auth,
```

Comment:
Just for my understanding, is there any way to restrict it to a list of allowed values without going the embedded schema route?

Reply:
you could write a `new` function or something and do some enforced enum checks there or something, but at that point it's probably easier just to make it an embedded schema

Reply:
```suggestion
                auth_type: :xml_auth,
```
this was the default before, we should probably keep it this way

Reply:
My POV was from the `Device.init` workflow where the initial auth_type will be `:no_auth` and then the `guess_auth()` finds the supported auth.

Reply:
Anyhow we are enforcing `:no_auth` for `GetSystemDateTime` call so we can keep any default.